src/jit/morph.cpp

```diff
@@ -1608,6 +1642,25 @@ void fgArgInfo::ArgsComplete()
                     // Spill multireg struct arguments that are expensive to evaluate twice
                     curArgTabEntry->needTmp = true;
                 }
+                else if (argx->OperGet() == GT_OBJ)
+                {
+                    GenTreeObj*           argObj     = argx->AsObj();
+                    CORINFO_CLASS_HANDLE  objClass   = argObj->gtClass;
+                    unsigned              structSize = compiler->info.compCompHnd->getClassSize(objClass);
+                    switch (structSize)
+                    {
+                    case 11:
+                    case 13:
+                    case 14:
+                    case 15:
```

Comment:
Added this comment:
                        // When we have a GT_OBJ of a struct with the above sizes we would need
                        // to use 3 or 4 load instructions to load the exact size of this struct.
                        // Instead we spill the GT_OBJ into a new GT_LCL_VAR temp and this sequence
                        // will use a GT_CPBLK to copy the exact size into the GT_LCL_VAR temp.
                        // Then we can just load all 16 bytes of the GT_LCL_VAR temp when passing
                        // the argument.
                        //